### PR TITLE
feat: User can set exact provider log level using environment variable

### LINKED ISSUE #2149: [➕ Feature]: Allow to change provider logger level via `PROVIDER_{PROVIDER_NAME}_LOG_LEVEL`
**Is your feature request related to a problem? Please describe.**
Sometimes we need to debug the particular provider and setting the global LOG_LEVEL will produce tons of logs. 

**Describe the solution you'd like**
The `PROVIDER_{PROVIDER_NAME}_LOG_LEVEL` will allow to toggle debug logs on selected providers.

### PR DESCRIPTION
**Pull request by VladimirFilonov**
Closes #2149 

## 📑 Description

Variable name:  `KEEP_PROVIDER_{}_LOG_LEVEL`

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

**Review by talboren — CHANGES_REQUESTED** [review]
I'm not sure why are we using this in BaseProvider:
`self.logger = context_manager.get_logger()`
and not
`self.logger = logging.getLogger(__name__)`
(I guess it's for automatic context, which can be achieved in other ways too)
But the easiest way to achieve what we're doing here in a "safer" way, is to have some way to add configuration to our logger via environment variables in `logging.py`

cc @shahargl @VladimirFilonov wdyt?

**Comment by talboren** [comment]
image

File: keep/contextmanager/contextmanager.py [hunk 1]
```diff
@@ -73,9 +75,23 @@ def api_key(self):
...
+        logger = logging.getLogger(name)
+        logger_adapter = WorkflowLoggerAdapter(
+            logger,
+            self,
+            self.tenant_id,
+            self.workflow_id,
+            self.workflow_execution_id,
+        )
+        self.__loggers.append(logger_adapter)
+        return logger_adapter
```
Comment (shahargl) [inline keep/contextmanager/contextmanager.py:94]: shouldn't it be smth like
```
self.__loggers = {}
..
if name in self.__loggers:
  return self.__loggers[name]
 
 self.__loggers[name] = logging.getLogger(name)
 return self.__loggers[name]
```

Comment (shahargl) [reply keep/contextmanager/contextmanager.py:94]: like, calling get_logger twice will init the the logger twice

File: tests/test_workflow_execution.py [hunk 2]
```diff
@@ -794,3 +845,184 @@ def wait_workflow_execution(workflow_id):
...
+    test_case,
+    alert_statuses,
+    expected_tier,
+    monkeypatch,
+):
+
+    logs_counts = {}
+    logs_level_counts = {}
+    for level in ["INFO", "DEBUG"]:
+        monkeypatch.setenv("KEEP_CONSOLE_PROVIDER_LOG_LEVEL", level)
```
Comment (shahargl) [inline tests/test_workflow_execution.py:971]: you can use the pytest.mark.parametrize to inject env var

```
@pytest.mark.parametrize(
    "test_app, test_case, alert_statuses, expected_tier, db_session",
    [
        ({"AUTH_TYPE": "NOAUTH", "KEEP_CONSOLE_PROVIDER_LOG_LEVEL": "INFO"}, "No action", [[0, "firing"]], None, None),
        ({"AUTH_TYPE": "NOAUTH", "KEEP_CONSOLE_PROVIDER_LOG_LEVEL": "DEBUG"}, "No action", [[0, "firing"]], None, None),
    ],
    indirect=["test_app", "db_session"],
)
```

Comment (shahargl) [reply tests/test_workflow_execution.py:971]: this will run the test twice instead of `for level in ["INFO", "DEBUG"]:`

Comment (talboren) [reply tests/test_workflow_execution.py:971]: @shahargl not sure how to share context between 2 different runs but the test checks that in debug mode there are more logs then info mode, wdyt?

**Review by shahargl — APPROVED** [review]
Lgtm

**Review by talboren — APPROVED** [review]
LGTM

**Pull request merged**